src/deephaven_riptable/riptable.py

```diff
@@ -0,0 +1,69 @@
+#
+#     Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+#
+"""This module provides the functions to support the conversion between Deephaven tables and Riptable Datasets."""
+from typing import List
+
+import deephaven.pandas as dhpd
+import deephaven.arrow as dhpa
+from deephaven import DHError
+from deephaven.table import Table
+
+try:
+    import riptable as rt
+except ImportError:
+    raise DHError(message="import riptable failed, please install riptable driver first.")
+
+
+def to_table(ds: rt.Dataset, cols: List[str] = None) -> Table:
+    """Creates a new table from a riptable Dataset.
+
+    Args:
+        ds (rt.Dataset): the riptable Dataset instance
+        cols (List[str]): the Dataset column names, default is None which means including all columns in the DataFrame
+
+    Returns:
+        a Deephaven table
+
+    Raise:
+        DHError
+    """
+    try:
+        pa_table = ds.to_arrow()
+        return dhpa.to_table(pa_table, cols)
+    except:
+        pass
+
+    try:
+        df = ds.to_pandas()
+        return dhpd.to_table(df, cols)
+    except DHError:
+        raise
+    except Exception as e:
+        raise DHError(e, message="failed to create a table from a riptable Dataset") from e
```

Comment:
Should we have an ABC instead, and allow the user to choose whether they want to to_table using arrow or pandas?

Reply:
Allowing the users to choose which intermediary format should be used is nice but may be a bit premature ATM. I think it is better to get some feedback from the users first.